core/src/main/java/com/scalar/db/api/Put.java

```diff
@@ -497,7 +498,18 @@ public Put withValue(Column<?> column) {
   @Deprecated
   public Map<String, Value<?>> getValues() {
     Map<String, Value<?>> ret = new HashMap<>();
-    columns.forEach((k, v) -> ret.put(k, v.hasNullValue() ? null : ScalarDbUtils.toValue(v)));
+    columns.forEach(
+        (k, c) -> {
+          Value<?> value;
+          if (c.hasNullValue()
+              && c.getDataType() != DataType.TEXT
+              && c.getDataType() != DataType.BLOB) {
```

Comment:
When a Text value and a Blob value are `null`, we should return `new TextValue("<column name>", null)` and `new BlobValue("<column name>", null)` respectively instead of `null`.